src/core/ext/filters/http/client/http_client_filter.cc

```diff
@@ -78,7 +78,10 @@ struct channel_data {
 
 static grpc_error* client_filter_incoming_metadata(grpc_call_element* elem,
                                                    grpc_metadata_batch* b) {
-  if (b->idx.named.status != nullptr) {
+  /* According to https://github.com/grpc/grpc/blob/master/doc/http-grpc-status-mapping.md
+   * we should prefer gRPC status codes over HTTP status codes if we have both.
+   */
+  if (b->idx.named.grpc_status != nullptr || b->idx.named.status != nullptr) {
```

Comment:
should be `b->idx.named.grpc_status == nullptr`, right?

Reply:
should've added work in progress tag :)